### PR TITLE
Update data selector to use remote data

### PR DESCRIPTION
**Pull request by alexandrudanpop**
Fixes OPS-1709

## Testing Checklist

Check all that apply:

- [x] I tested the feature thoroughly, including edge cases

- [x] I verified all affected areas still work as expected

- [x] Automated tests were added/updated if necessary

- [x] Changes are backwards compatible with any existing data, otherwise a migration script is provided

## Visual Changes (if applicable)

Deploying to UX
https://www.loom.com/share/6dd037ffaae94403bc49fd9f35f7f771

### CODE REVIEW

File: packages/server/api/src/app/flags/flag.service.ts [hunk 1]
```diff
@@ -290,7 +290,7 @@ export const flagService = {
       },
       {
         id: FlagId.USE_NEW_EXTERNAL_TESTDATA,
-        value: false,
+        value: true,
```
Comment (alexandrudanpop) [inline packages/server/api/src/app/flags/flag.service.ts:293]: todo reset

**Review by rSnapkoOpenOps — APPROVED** [review]

**Review by cezudas — APPROVED** [review]
lgtm
